developer/src/kmc-kmn/src/compiler/compiler.ts

```diff
@@ -107,4 +111,72 @@ export class Compiler {
       return false;
     }
   }
-}
\ No newline at end of file
+
+  /**
+   *
+   * @param pattern UnicodeSet pattern such as `[a-z]`
+   * @param bufferSize guess as to the buffer size
+   * @returns UnicodeSet accessor object
+   */
+  public async parseUnicodeSet(pattern: string, bufferSize: number) : Promise<UnicodeSet> {
+    if (!bufferSize) {
+      bufferSize = 100;
+    }
+
+    const initOk = await this.init();
+    if (!initOk) {
+      throw Error(`WASM machinery didn't start up properly`);
+    }
+    // Module seems to be the usual name for it
+    const Module = this.wasmModule;
+    const buf = Module.asm.malloc(bufferSize * 2 * Module.HEAPU32.BYTES_PER_ELEMENT);
+    const rc = this._parseUnicodeSet(pattern, buf, bufferSize);
+    if (rc < 0) {
+      throw new UnicodeSetError(rc);
```

Comment:
I think we want to be reporting through the normal `reportMessage` callback and defining a Error_ message

Reply:
yes